src/ecr.sw

```diff
@@ -0,0 +1,54 @@
+library ecr;
+
+use ::b512::B512;
+use ::address::Address;
+use ::chain::assert;
+
+/// Recover the public key derived from the private key used to sign a message
+pub fn ec_recover(signature: B512, msg_hash: b256) -> B512 {
+    let public_key = ~B512::new();
+
+    public_key.hi = asm(buffer, hi: signature.hi, hash: msg_hash) {
+        move buffer sp; // Result buffer.
+        cfei i32;
+        ecr buffer hi hash;
+        buffer: b256
+    };
+    // @note temp
+    assert(public_key.hi == 0x1d152307c6b72b0ed0418b0e70cd80e7f5295b8d86f5722d3f5213fbd2394f36);
+
+    // a `B512`'s 2 internal values are stored in contiguous memory. This asm block takes a pointer to the initial value (public_key.hi) and should retrieve the lo value from memory and return it to be stored in public_key.lo.
+    public_key.lo = asm(buffer, hi_ptr: public_key.hi, lo_ptr) {
```

Comment:
The asm block above works as expected, and the assert passes.
However, this block is not working correctly. I'm pretty certain the issue lies in my usage of assembly to try the recover the lo bits of the B512 value from memory.

Reply:
The `assert` below is failing,